ahnlich/ahnlich_ai/src/error.rs

```diff
@@ -0,0 +1,21 @@
+use ahnlich_types::keyval::StoreName;
+use ahnlich_types::metadata::MetadataKey;
+use thiserror::Error;
+
+/// TODO: Move to shared rust types so library can deserialize it from the TCP response
+#[derive(Error, Debug, Eq, PartialEq, PartialOrd, Ord)]
+pub enum AIProxyError {
+    #[error("Predicate {0} not found in store, attempt CREATEINDEX with predicate")]
+    PredicateNotFound(MetadataKey),
+    #[error("Store {0} not found")]
+    StoreNotFound(StoreName),
+    #[error("Store {0} already exists")]
+    StoreAlreadyExists(StoreName),
+    #[error("Store dimension is [{store_dimension}], input dimension of [{input_dimension}] was specified")]
+    StoreDimensionMismatch {
```

Comment:
Also don't need this in AI... we are the ones handling the store dimension via the model so it should always be consistent